staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -172,6 +173,13 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("listKind"), names.ListKind, "kind and listKind may not be the same"))
 	}
 
+	// kind cannot be a top-level type since they are shared across all groupversions
+	for _, topLevelType := range scheme.Scheme.KnownTypes(scheme.UnversionedVersion) {
```

Comment:
is it concerning that addition of a new top-level type in the future could cause existing CRDs to fail validation? should we handle this as a name conflict reflected in status acceptedNames instead?

Reply:
The cure is to use meta.k8s.io/v1 for all those types, not magically add them to every scheme.

Reply:
> The cure is to use meta.k8s.io/v1 for all those types, not magically add them to every scheme.

True. Is the list this looks at set up so that it will never expand, and new types will be properly grouped that way?

Reply:
> and new types will be properly grouped that way?

We have to watch any new type to be grouped. This would make sense to be added to the api conversions doc in the community repo.